modules/core/src/convert.cpp

```diff
@@ -4585,7 +4585,7 @@ cvtScaleHalf_<float, short>( const float* src, size_t sstep, short* dst, size_t
                     __m128i v_dst = _mm_cvtps_ph(v_src, 0);
 
                     _mm_storel_epi64((__m128i *)(dst + x), v_dst);
-#elif defined __GNUC__ && (defined __arm__ || defined __aarch64__)
+#elif defined __GNUC__ && (defined CV_NEON || defined __aarch64__)
```

Comment:
Curious....
float16x4_t is usually defined in arm_neon.h.

```
odroid@odroid:~$ ls /usr/lib/gcc/arm-linux-gnueabihf/4.8.4/include/arm_neon.h
/usr/lib/gcc/arm-linux-gnueabihf/4.8.4/include/arm_neon.h
odroid@odroid:~$ grep float16x4_t !$
grep float16x4_t /usr/lib/gcc/arm-linux-gnueabihf/4.8.4/include/arm_neon.h
typedef __builtin_neon_hf float16x4_t   __attribute__ ((__vector_size__ (8)));
```

It depends on the compiler, so I was suspecting that this definition was guarded by ```#if ((__ARM_FP & 0x2) != 0``` and __ARM_FP has 12 (i.e. mfpu=neon is accepted, but not the mfpu=neon-fp16)
In @alalek log, it states that ```__ARM_FP``` is 14 and ```__ARM_NEON_FP``` is 6, which means, mfpu=neon-fp16 is clearly passed.
I wonder why.